test/core/VarReplay.jl

```diff
@@ -94,68 +94,20 @@ include("../test_utils/AllUtils.jl")
         randr(vi, vn_z3, dists[1], spl1)
         @test vi.orders == [1, 1, 2, 2, 2, 3]
         @test vi.num_produce == 3
-        @test getretain(vi, spl1) == UnitRange[]
 
-        # Check getretain at different stages
-        # variables samples in different order: z1,a1,z2,z3,a2
         vi.num_produce = 0
-        @test getretain(vi, spl1) == UnitRange[6:6,5:5,4:4,2:2,1:1]
-        @test getretain(vi, spl2) == UnitRange[3:3]
         set_retained_vns_del_by_spl!(vi, spl1)
 
         vi.num_produce += 1
         randr(vi, vn_z1, dists[1], spl1)
         randr(vi, vn_a1, dists[2], spl1)
-        @test getretain(vi, spl1) == UnitRange[4:4,5:5,6:6]
         vi.num_produce += 1
         randr(vi, vn_z2, dists[1], spl1)
-        @test getretain(vi, spl1) == UnitRange[6:6]
         vi.num_produce += 1
         randr(vi, vn_z3, dists[1], spl1)
         randr(vi, vn_a2, dists[2], spl1)
         @test vi.orders == [1, 1, 2, 2, 3, 3]
         @test vi.num_produce == 3
-
```

Comment:
why we are removing these tests?

Reply:
They only test `getretain` which is not used anywhere in Turing, so I removed it.

Reply:
I think this was originally used by the Gibbs sampler/particle Gibbs sampler. 

@xukai92 do we have a replacement for `getretrain` now?

Reply:
Just checked the code. It seems that the functionality is replaced by `is_flagged(vi, vn, "del")`.

At each PG step or when task is coped, `set_retained_vns_del_by_spl!(vi, spl)` is triggered to flag all variables for the sampler as `del`.

When the model is run, `assume` function behave depends on `is_flagged(vi, vn, "del")` and whether or not the variable belongs to the sampler.

The logic looks correct to me.

I'm not sure if it's me or @emilemathieu did this change.